### PR TITLE
Add 7 Matrix Unit Tests and Implement Matrix Multiplication

### LINKED ISSUE #468: Matrix multiplication
**Why?**
very important function used almost in every graphics application
**How?**
there is a simple O(n^3) solution (with possibility of Strassen's multiplication)
**Would I work on this issue**
yes

### PR DESCRIPTION
**Pull request by youdymoo**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

- [X] **If creating a new file:**
  - [X] added links to it in the README files?
  - [X] included tests with it? (in tests/test_matrix.py)
  - [X] added description (overview of algorithm, time and space complexity, and possible edge case) in docstrings?

- [X] **if done some changes:**
  - [X] wrote a short description in the PR explaining what the changes do?
  - [X] fixes #468  if related to any issue

- [X] **other**
  - [X] add valid unit tests of 7 methods of matrix (former coverage only 5.25%, now 49.86%)
  - [X] follows the PEP8 code style

### CODE REVIEW

File: tests/test_matrix.py [hunk 1]
```diff
@@ -38,9 +90,98 @@ def test_crout_matrix_decomposition(self):
...
+                    [
+                        5, 3, 4, 6, 7, 8, 9, 1, 2], [
+                        6, 7, 2, 1, 9, 5, 3, 4, 8], [
+                        1, 9, 8, 3, 4, 2, 5, 6, 7], [
+                            8, 5, 9, 7, 6, 1, 4, 2, 3], [
+                                4, 2, 6, 8, 5, 3, 7, 9, 1], [
+                                    7, 1, 3, 9, 2, 4, 8, 5, 6], [
+                                        9, 6, 1, 5, 3, 7, 2, 8, 4], [
+                                            2, 8, 7, 4, 1, 9, 6, 3, 5], [
+                                                3, 4, 5, 2, 8, 6, 1, 7, 9]]))
```
Comment (keon) [inline tests/test_matrix.py]: can you align the columns?

File: algorithms/matrix/multiply.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    # create a result matrix
+    result = [[0] * multiplier_col for i in range(multiplicand_row)]
+    for i in range(multiplicand_row):
+        for j in range(multiplier_col):
+            for k in range(len(multiplier)):
+                result[i][j] += multiplicand[i][k] * multiplier[k][j]
+    return result
+
+
+# def main():
```
Comment (keon) [inline algorithms/matrix/multiply.py]: forgot to remove these?

**Comment by keon** [comment]
thanks for the contribution! this will be a nice addition to the matrix related algo.
i left some minor comments :)

File: tests/test_matrix.py (hunk 1 above)
Comment (youdymoo) [reply tests/test_matrix.py]: Sure. It's auto-formatted by the autopep8 LOL

**Comment by youdymoo** [comment]
Just updated! Thanks for the review.
